kvdb-rocksdb/src/lib.rs

```diff
@@ -518,20 +518,20 @@ impl Database {
 			let mut read_opts = ReadOptions::default();
 			read_opts.set_verify_checksums(false);
 			let end_prefix = kvdb::end_prefix(prefix).into_boxed_slice();
-			// SAFETY: the end_prefix lives as long as the iterator
-			// See `ReadGuardedIterator` definition for more details.
-			unsafe {
-				read_opts.set_iterate_upper_bound(&end_prefix);
+			// rocksdb doesn't work with an empty upper bound
+			if !end_prefix.is_empty() {
```

Comment:
cc @cheme, we might want to re-review prefix deletion code as rocksdb doesn't iterate at all if the upper bound is empty as it matches all keys

Reply:
I should have remember that, there was some similar things in the deletion prefix pr, let me check back:
https://github.com/paritytech/parity-common/blob/dd89c9a00b82b9e5c4e18964fd452c0dca5eed3c/kvdb-rocksdb/src/lib.rs#L447
Is there a case where we got a empty upper bound and the start bound is not empty, ok [255] or [255, 255] ... So yes the test should be rewrite on the result of end prefix instead of the prefix :+1: I am doing a quick PR.